### PR TITLE
Add NECAT

### PR DESCRIPTION
**Pull request by natir**
Add Necat, a nanopore long read assembler.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by natir** [comment]
@BiocondaBot please fetch artifacts

**Comment by natir** [comment]
@BiocondaBot please add label

**Comment by natir** [comment]
@BiocondaBot please update

**Comment by natir** [comment]
@BiocondaBot please add label
@BiocondaBot please fetch artifacts

**Comment by natir** [comment]
@BiocondaBot please add label

**Comment by natir** [comment]
I create a pull request in NECAT repo to include patch I create xiaochuanle/NECAT#25

**Comment by natir** [comment]
@innovate-invent this is the recipe with many script and binary we discussed on Gitter.

File: recipes/necat/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -ex
+
+export CFLAGS="$CFLAGS  -I$PREFIX/include"
+export CXXFLAGS="$CXXFLAGS  -I$PREFIX/include"
+export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
+
+# build
+cd src
+make CFLAGS+="${CFLAGS}" LDFLAGS+="${LDFLAGS}" CXXFLAGS+="${CXXFLAGS}"
```
Comment (innovate-invent) [inline recipes/necat/build.sh:10]: Replace `cd` with -C argument for make

File: recipes/necat/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+cd ..
+
+# create and populate binary file
+mkdir -p $PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/
```
Comment (innovate-invent) [inline recipes/necat/build.sh:14]: The FSH standard requires the path to be `/usr/share`

File: recipes/necat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - {{ compiler("cxx") }}
+  host:
+    - zlib
+  run:
+    - perl
+    - perl-env
+
+test:
+  commands:
+    - necat
```
Comment (innovate-invent) [inline recipes/necat/meta.yaml:69]: pipe to grep to ensure expected output ie `necat | grep 'some expected output'
grep will return non-zero if it is not found

File: recipes/necat/build.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+cp -r Linux-amd64/bin/*  $PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/
+
+# create calling script
+mkdir $PREFIX/bin/
+cat <<EOF > $PREFIX/bin/necat
+#!/bin/bash
+
+PATH=\$PATH:$PREFIX/share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/bin/ necat.pl \$@
```
Comment (innovate-invent) [inline recipes/necat/build.sh:22]: double quote $@, ie `"$@"`

**Review by innovate-invent — CHANGES_REQUESTED** [review]
Just a few minor things, otherwise great work.

File: recipes/necat/build.sh (hunk 2 above)
Comment (natir) [reply recipes/necat/build.sh:14]: Her I follow the [java recommendation](https://bioconda.github.io/contributor/guidelines.html?highlight=java#java) it's maybe not a good idea.

File: recipes/necat/build.sh (hunk 4 above)
Comment (natir) [reply recipes/necat/build.sh:22]: done

File: recipes/necat/meta.yaml (hunk 3 above)
Comment (natir) [reply recipes/necat/meta.yaml]: done

File: recipes/necat/build.sh (hunk 1 above)
Comment (natir) [reply recipes/necat/build.sh]: done

**Comment by natir** [comment]
I didn't understand why build failed now

**Comment by natir** [comment]
Ok don't put `:` in test string

File: recipes/necat/build.sh (hunk 2 above)
Comment (innovate-invent) [reply recipes/necat/build.sh:12]: ah, I see. Never mind then.

**Review by innovate-invent — APPROVED** [review]

**Comment by natir** [comment]
@BiocondaBot please merge

**Pull request merged**